### ISSUE TITLE
SC1087 false negative (postitive?)

### ISSUE THREAD

**Issue by mmlb** [koalaman/shellcheck#1212]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC1087
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

makers=(1 2)
for maker in "${makers[@]}"; do
    echo "maker=$maker[]"
done
```

#### Here's what shellcheck currently says:
```
Line 5	SC1087: Use braces when expanding arrays, e.g. ${array[idx]} (or ${var}[.. to quiet).
```
#### Here's what I wanted or expected to see:

No errors

_side note: awesome pre-filled report from shellcheck.net_

**Comment by matthewpersico** [comment]
FWIW, this gives no errors in shellcheck.

```
#!/bin/bash

makers=(1 2)
for maker in "${makers[@]}"; do
    echo "maker=${maker}[]"
done
```

**Comment by mmlb** [comment]
right, I think the issue is that shellcheck is incorrectly thinking that [] is array dereference, when really its just open/close square bracket literals

**Comment by koalaman** [comment]
Shellcheck was incorrectly thinking that maybe *you* thought it would be an array reference.

Perl and PHP both use `$foo[bar]` to access elements in an array, so this is not always an obvious mistake.

I agree that the wiki worded this strangely, and I've updated it. Thanks!

**Comment by mmlb** [comment]
:+1:
